plugin/al/usdmayautils/AL/usdmaya/utils/MeshUtils.cpp

```diff
@@ -751,8 +759,171 @@ bool MeshImportContext::applyEdgeCreases()
+        
+                    for (uint32_t i = 0, n = rawValColour.size(); i < n; ++i)
+                      colours[i] = MColor(rawValColour[i][0], rawValColour[i][1], rawValColour[i][2], rawValOpacity[i]);
+                    representation = MFnMesh::kRGBA;
+                    setCombinedDisplayAndOpacityColourSet = true;
+                  }
+                }
+              }
+            }
+            if (!setCombinedDisplayAndOpacityColourSet) 
+            {
+              const VtArray<GfVec3f> rawVal = vtValue.UncheckedGet<VtArray<GfVec3f> >();
+              colours.setLength(rawVal.size());
+              for (uint32_t i = 0, n = rawVal.size(); i < n; ++i)
+                colours[i] = MColor(rawVal[i][0], rawVal[i][1], rawVal[i][2]);
+              representation = MFnMesh::kRGB;
+            }
+          }
+          else if (vtValue.IsHolding<VtArray<GfVec4f> >())
+          {
+            const VtArray<GfVec4f> rawVal = vtValue.UncheckedGet<VtArray<GfVec4f> >();
+            colours.setLength(rawVal.size());
+            memcpy(&colours[0], (const float*) rawVal.cdata(), sizeof(float) * 4 * rawVal.size());
+            representation = MFnMesh::kRGBA;
+          }
+        
+          // Set colors
+          if (!fnMesh.setColors(colours, &colourSetName, representation))
+          {
+            TF_DEBUG(ALUTILS_INFO).Msg("Failed to set colours for colour set \"%s\" on mesh \"%s\", error: %s\n",
+                colourSetName.asChar(), fnMesh.name().asChar(), s.errorString().asChar());
+            continue;
+          }
+
+          // When primvar is indexed assume these indices
+          MIntArray mayaIndices;
+          VtIntArray usdindices;
+          if (primvar.GetIndices(&usdindices, m_timeCode))
+          {
+            mayaIndices.setLength(usdindices.size());
+            std::memcpy(&mayaIndices[0], usdindices.cdata(), sizeof(int) * usdindices.size());
+            if (mayaIndices.length() != connects.length())
+            {
+              TF_DEBUG(ALUTILS_INFO).Msg(
+                  "Retrieved indexed values are not compatible with topology for colour set \"%s\" on mesh \"%s\"\n",
+                  colourSetName.asChar(), fnMesh.name().asChar());
+              continue;
+            }
+          }
+
+          // Otherwise generate indices based on interpolation
+          if (mayaIndices.length() == 0)
+          {
+            if (interpolation == UsdGeomTokens->faceVarying)
+            {
+              generateIncrementingIndices(mayaIndices, colours.length());
+            }
+            else if (interpolation == UsdGeomTokens->uniform)
+            {
+              if (colours.length() == counts.length())
+              {
+                mayaIndices.setLength(connects.length());
+                for (uint32_t i = 0, idx = 0, n = counts.length(); i < n; ++i)
+                {
+                  for (uint32_t j = 0; j < counts[i]; ++j)
```

Comment:
error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
`for (uint32_t j = 0; j < counts[i]; ++j)`